pkg/sql/resolver.go

```diff
@@ -722,6 +722,17 @@ func (p *planner) getTableAndIndex(
 		return nil, nil, err
 	}
 	optIdx := idx.(*optIndex)
+
+	// Resolve the object name for logging if
+	// its missing.
+	if tableWithIndex.Table.ObjectName == "" {
+		tableName, err := p.getQualifiedTableName(ctx, optIdx.tab.desc)
+		if err != nil {
+			return nil, nil, err
+		}
+		tableWithIndex.Table = *tableName
+	}
```

Comment:
This might do the job, nonetheless there are a few things worth pointing out:
- calling `getQualifiedTableName` comes at a cost; looking up the db and schema descriptors might end up doing so from storage, even if it's unlikely;
- if `ResolveTableIndex` was able to find the index of a table, then it seems to me that it should have been able to  resolve that table in the first place. The reason I suspect this is because in CRDB, indexes are always defined within a table descriptor and can't exist independently. Looking at the signature of `ResolveTableIndex`, I further suspect that it already returns that qualified table name as its second argument.